code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -5234,7 +5233,7 @@
 		M.setGender(FEMALE)
 	if(ishuman(M))
 		var/mob/living/carbon/human/H = M
-		if(!istype(H.w_uniform,/obj/item/clothing/under/schoolgirl))
+		if(!M.is_wearing_item(/obj/item/clothing/under/schoolgirl))
```

Comment:
you need to specify the correct slot I believe, otherwise `is_wearing_item` will return true if you're holding the dress in your hand or something.